static/app/utils/replays/fetchReplayList.tsx

```diff
@@ -59,7 +59,7 @@ async function fetchReplayList({
         // when queryReferrer === 'issueReplays' we override the global view check on the backend
         // we also require a project param otherwise we won't yield results
         queryReferrer,
-        project: queryReferrer === 'issueReplays' ? ALL_ACCESS_PROJECTS : payload.project,
```

Comment:
This wasn't doing anything before. We only call `fetchReplayList` from the hook `useReplayList`... and that hook is only used in 4 places, all are updated in this PR to have specific referrers now.

Also the types prevented any other values from appearing here.

Reply:
and... this might actually be doing something on the backend.

Reply:
Yup, right here: https://github.com/getsentry/sentry/blob/c5d004cb0f0c94547df196e29f4d9fb0b53daa14/src/sentry/replays/endpoints/organization_replay_index.py#L41-L44